src/components/reservations/Filters.js

```diff
@@ -0,0 +1,203 @@
+        control={<Switch checked={formValues.isElectric} onChange={onIsElectricSwitchChange} name="isElectric" />}
+        label="Electric Engine"
+      />
+
+      <FormControl>
+        <InputLabel id="frame-size-select">Frame Size</InputLabel>
+        <Select
+          labelId="frame-size-select"
+          id="frame-size-select"
+          value={formValues.selectedFrameSize}
+          onChange={onFrameSizeChange}
+        >
+          {[...frameSizes].map((size) => setFrameSizePickListElement(size))}
+        </Select>
+      </FormControl>
+      <FormControl>
+        <InputLabel id="rent-from-select">Pickup location</InputLabel>
+        <Select
+          labelId="rent-from-select"
+          id="rent-from-select"
+          value={formValues.rentFrom}
+          onChange={onPickupLocationChange}
+        >
+          {[...rentalPoints].map((point) => setRentalPointLocationPickListElement(point))}
+        </Select>
+      </FormControl>
+      <FormControl required error={inputErrors.rentTo}>
+        <InputLabel id="rent-to-select">Return Location</InputLabel>
+        <Select
+          labelId="rent-to-select"
+          id="rent-to-select"
+          value={formValues.rentTo}
+          onChange={onReturnLocationChange}
+        >
+          {[...rentalPoints].map((point) => setRentalPointLocationPickListElement(point))}
+        </Select>
+        {inputErrors.rentTo && <FormHelperText>Select return location</FormHelperText>}
+      </FormControl>
+
+      <MuiPickersUtilsProvider utils={DateFnsUtils}>
+        <FormControl required error={inputErrors.startDate}>
+          <DateTimePicker
+            label="Reservation start date"
+            value={formValues.startDate}
+            onChange={onStartDateChange}
+            disablePast
+            showTodayButton
+            ampm={false}
+            autoOk={true}
+            format="dd MMMM yy HH:MM"
+            margin="normal"
+            inputVariant="standard"
+          />
+          {inputErrors.startDate && <FormHelperText>Select date</FormHelperText>}
+        </FormControl>
+        <FormControl required error={inputErrors.endDate}>
+          <DateTimePicker
+            label="Reservation end date"
+            value={formValues.endDate}
+            onChange={onEndDateChange}
+            disablePast
+            minDate={formValues.startDate}
+            ampm={false}
+            autoOk={true}
+            format="dd MMMM yy HH:MM"
+            margin="normal"
+            inputVariant="standard"
+          />
+          {inputErrors.endDate && <FormHelperText>Select date</FormHelperText>}
+        </FormControl>
+      </MuiPickersUtilsProvider>
+      <FormControl error={inputErrors.selectedBikes}>
+        <Button variant="contained" color="primary" onClick={handleClick}>
+          Create Reservation
+        </Button>
+        {inputErrors.selectedBikes && <FormHelperText>Select bike</FormHelperText>}
+      </FormControl>
+    </Box>
+  );
+};
```

Comment:
Same as above - too many things in one file. But focus on the logic, let Mikołaj do the refactor.

Reply:
ok